File: Лаборатори/Lab3-sub.java
class Solution {
    public String longestNiceSubstring(String s) {
        if (s.length() < 2) {
            return "";
        }
        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            
            if (s.indexOf(Character.isLowerCase(ch) ? Character.toUpperCase(ch) : Character.toLowerCase(ch)) == -1) {
                String left = longestNiceSubstring(s.substring(0, i));
                String right = longestNiceSubstring(s.substring(i + 1));
        
                return left.length() >= right.length() ? left : right;
            }
        }
        return s;
    }
}

File: Лаборатори/Lab3-sub.py
class Solution:
    def longestNiceSubstring(self, s: str) -> str:
        
        if len(s) < 2:
            return ""
        
        for i, char in enumerate(s):
            if char.swapcase() not in s:
                
                left = self.longestNiceSubstring(s[:i])
                right = self.longestNiceSubstring(s[i+1:])
                
                return left if len(left) >= len(right) else right
        return s